blog/content/post/2022-12-16-trenchboot-aem-for-qubesos.md

```diff
@@ -0,0 +1,367 @@
+---
+title: TrenchBoot Anti Evil Maid for Qubes OS
+abstract: 'Qubes OS Anti Evil Maid (AEM) software heavily depends on the
+           availability of the DRTM technologies to prevent the Evil Maid
+           attacks. However, the project hasn't evolved much since the
+           beginning of 2018 and froze on the support of TPM 1.2 with Intel TXT
+           in legacy boot mode (BIOS). In the post we show how existing
+           solution can be replaced with TrenchBoot and how one can install it
+           on the Qubes OS. Also the post will also briefly explain how
+           TrenchBoot opens the door for future TPM 2.0 and UEFI support for
+           AEM.'
+cover: /covers/trenchboot-logo.png
+author: michal.zygowski
+layout: post
+published: true
+date: 2022-12-16
+archives: "2022"
+
+tags:
+  - firmware
+  - coreboot
+  - Qubes OS
+  - TrenchBoot
+  - GRUB
+  - Xen Hypervisor
+categories:
+  - Firmware
+  - Bootloader
+  - Hypervisor
+  - OS Dev
+  - Security
+
+---
+
+# Introduction
+
+The firmware is the heart of the security of a given system and should always
+be up-to-date to maintain the computer's security. However, being up to date
```

Comment:
There can be more reasons e.g. regulatory compliance, but I guess this can be treated as feature critical to user. Firmware update should be released when it provide value otherwise what is the point for new firmware release.

OTOH @krystian-hebel based on your assumption and considering quality of release notes for most firmware releases firmware, which is primary source of knowledge about update content, user should almost never updated, because how you can judge if "updated performance" is of any value for you.

There are some users who update just to check if anything in their specific workflow was improved.

Finally some hardware warranty or standard operation procedures may have written it should always operate with most recent update. In such case users has not much to discuss then comply - again this can be in feature-critical-to-user bucket. So that bucket is pretty big.

Reply:
This is not a place for ideological discussions. Please suggest change or close the thread and move to Matrix or something.  I will just leave a comment on this:

> considering quality of release notes for most firmware releases firmware, which is primary source of knowledge about update content

Even if release notes do not mention something , it doesn't mean that some bugs have been silently patched. I have wrote this paragraph with such interpretation in mind.

Reply:
My comment is by no means blocking. You made very generic statement with which @krystian-hebel disagree. Maybe introduction of this post is not the place for such generic claims and those generic claims should be removed and we should focus on what we would like to present further in the blog post? Maybe we should think to who we target this post (I mean Qubes OS users).

e.g.

As Qubes OS users, promoters, and developers, we understand how essential it is to be aware of the latest developments in maintaining the security of your favorite operating system. We're excited to share our plans to integrate the TrenchBoot Project into Qubes OS's new Anti-Evil Maid (AEM) implementation. As you may know, traditional firmware security measures like UEFI Secure Boot and measured boot, even with a Static Root of Trust (SRT), may only sometimes be enough to ensure a completely secure environment for your operating system. Compromised firmware may allow for the injection of malicious software into your system, making it difficult to detect. One recent example was the misconfiguration of UEFI Secure Boot on MSI platforms. To overcome these limitations, many silicon vendors have started implementing Dynamic Root of Trust (DRT) technologies to establish a secure environment for operating system launch and integrity measurements. We're excited to take advantage of these advancements through integration with the TrenchBoot Project.

Reply:
> Maybe introduction of this post is not the place for such generic claims and those generic claims should be removed and we should focus on what we would like to present further in the blog post? Maybe we should think to who we target this post (I mean Qubes OS users).

Ok, I just wanted to make my job to be easier and cited my original words from abstract: https://docs.dasharo.com/projects/trenchboot-aem/#abstract
But I guess I can't lesser my efforts. At this point I must ask @krystian-hebel to suggest changes to this paragraph so that he likes it.

Reply:
I like what @pietrushnic wrote, just pass it through grammarly first.

Reply:
Done

Reply:
LGTM

Reply:
@krystian-hebel and @pietrushnic agreed - I assume this is resolved.